mautic/mautic
# Send channel ID to sendgrid so we can use it in the bounce webhook

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ yes]
| New feature/enhancement? (use the a.x branch)      | [no ]
| Deprecations?                          | [ none]
| BC breaks? (use the c.x branch)        | [ none]
| Automated tests included?              | [ not yet] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #9045 



#### Description:
This fixes #9045. It sends the channel ID to sendgrid as an extra custom argument, which is then read during the handling of the callback for bounced/failed emails.




#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Select Sendgrid API as the mail service
3. Create a segment and add a contact that has an email address that will bounce
4. Create an email channel for the segment
5. Hit send for the email channel
6. You should now see a bounce inside the email channel graph




<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/11052"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
